### PR TITLE
[Merged by Bors] - chore: Move `WithZero` material depending on `GroupWithZero`

### PR DESCRIPTION
**Pull request by YaelDillies**
Everything under `Algebra.Group` should be additivisable. Therefore I move the `GroupWithZero` instances for `WithZero` from `Algebra.Group.WithOne.Defs` and the whole of `Algebra.Group.WithOne.Units` to a new file `Algebra.GroupWithZero.WithZero`. I credit Mario for https://github.com/leanprover-community/mathlib/commit/ad92a9ba47f417916aab365d13db653fa8991a84 and Johan for https://github.com/leanprover-community/mathlib/pull/762.

Use the opportunity to slightly clean up the code:
* Replace `:=` by `where` in instance declarations
* Add missing explicit arguments to `coe` lemmas
* Add missing `section ... end`

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by eric-wieser** [comment]
On the assumption that you did more than just moves (eg, replacing `{}` with `where`), can you summarize the changes you made in the PR description?

**Comment by YaelDillies** [comment]
Done

**Comment by eric-wieser** [comment]
bors d+

Thanks!

**Comment by YaelDillies** [comment]
bors merge
